provider/execution/execution.go

```diff
@@ -407,6 +414,14 @@ func (e TableExecutor) resolveColumns(ctx context.Context, meta schema.ClientMet
 		if c.Resolver != nil {
 			e.Logger.Trace("using custom column resolver", "column", c.Name)
 			err := c.Resolver(ctx, meta, resource, c)
+			// if default wasn't set as nil and our resource is either nil or we got an error we set the default value
+			if c.Default != nil && (err != nil || resource.Get(c.Name) == nil) {
```

Comment:
Doesn't work right now for creating an empty json for `aws_lambda_functions.Tags`.

In the golang struct from the `aws-sdk-for-go-v2`, There is a `nil` value in the `tags` field (the field is of type `map[string]string`. But in the code we converted to `interface{}`, so it becomes a `nil` of type `map[string]string`. This means that in the code when we check if `resource.Get(c.Name)` == nil, we get `false` because it's not `nil`, it's "nil of type map[string]string". Thus we don't set a default value and instead get a json-null, instead of a json-empty-objects.

Some sceenshots:

<img width="1725" alt="image" src="https://user-images.githubusercontent.com/37939765/171635279-2fb5d57a-bcb4-4144-af19-36ca8d431f0f.png">

<img width="851" alt="image" src="https://user-images.githubusercontent.com/37939765/171635424-796bb966-8d61-48c6-832c-29a3c641c5c0.png">

<img width="851" alt="image" src="https://user-images.githubusercontent.com/37939765/171635839-7ec7b354-996b-4c0c-8acc-1af7e5c027ed.png">

<img width="1347" alt="image" src="https://user-images.githubusercontent.com/37939765/171635939-70133f24-7e6f-4238-bcd7-2aeced1f6331.png">

Reply:
Rather than making the code even more complicated, e.g. with `reflect2.IsNil`, I propose to rethink whether we can just not implement this feature.

Took me 3 hours of testing to find this bug, and I am not convinced I tried all the possible configurations of nulls, types, and null-types... 

postesql nulls, nulls of type `interface{}`, nulls of other types contained in a `interface{}` (which is not null itself..)..

Reply:
Maybe just an ad-hoc resolver for this specific function? 
The relevant issue https://github.com/cloudquery/cloudquery-issues/issues/358 just mentions 2 specific places where we don't handle nulls correctly..